pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -44,6 +44,18 @@
     OMEGACONF_AVAILABLE = True
 
 
+# [Save & Load scheme]
```

Comment:
It is a great idea, but I have some trouble to understand what you meant to explain. Would mind refactoring this comment ?

Reply:
What I hope to explain is "how states is saved, then loaded".  

- save: various states -> checkpoint dict -> .ckpt file
- load: .ckpt file -> checkpoint dict -> various states 

And also hope to explain "which function implement each steps".  

- save: `save_checkpoint` (`dump_checkpoint` for making dict & `atomic_save` for write) 
- load: `restore` (`pl_load` for read & part of `restore` for restoring states)

But the scheme graph looks complex as you mentioned.  
**I will simply write above comment in each function.**

Reply:
Refactored.